# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GCAPI.cpp
# Commit: 73928903648a
# Full Hash: 73928903648aa79ca0b4fa0bb27ae844c34577e5
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-08-24 21:34:05
# Regressor Bug: 1786506
# File Overlap Count: 1
# Description:
#   Bug 1786506 - Part 2: Give shared permanent things their own zone r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155378
# ==============================================================================

diff -r 747154213937 -r 73928903648a js/src/gc/GCAPI.cpp
--- a/js/src/gc/GCAPI.cpp	Wed Aug 24 09:43:06 2022 +0000
+++ b/js/src/gc/GCAPI.cpp	Wed Aug 24 09:43:06 2022 +0000
@@ -224,8 +224,14 @@
 JS_PUBLIC_API void JS::PrepareZoneForGC(JSContext* cx, Zone* zone) {
   AssertHeapIsIdle();
   CHECK_THREAD(cx);
+
+  // If we got the zone from a shared atom, we may have the wrong atoms zone
+  // here.
+  if (zone->isAtomsZone()) {
+    zone = cx->runtime()->atomsZone();
+  }
+
   MOZ_ASSERT(cx->runtime()->gc.hasZone(zone));
-
   zone->scheduleGC();
 }
 